### PR TITLE
Table and events

### PR DESCRIPTION
**Pull request by ArteGon**
Work with table and events, also add table pagination, which work with backend

### CODE REVIEW

File: src/components/common/table/table.tsx [hunk 1]
```diff
@@ -182,8 +191,10 @@ const Table: FunctionComponent<TableProps> = ({
   const { allColumns, headerGroups, getTableBodyProps, page, prepareRow, state } = instance;
 
   useEffect(() => {
-    instance.state.hiddenColumns = allColumns.filter(column => column.hasOwnProperty('startHide')).map(column => column.id)
-  }, [])
+    instance.state.hiddenColumns = allColumns
+      .filter(column => column.hasOwnProperty('startHide'))
+      .map(column => column.id);
```
Comment (ZmicerBoksha) [inline src/components/common/table/table.tsx]:
```suggestion
      .map({ id } => id);
```

File: src/components/common/table/tablePagination/tablePagination.tsx [hunk 2]
```diff
@@ -13,43 +12,43 @@ const useStyles = makeStyles(() => {
...
-  const {
-    state: { pageIndex, pageSize },
-    setPageSize,
-  } = instance;
-
-  useEffect(() => {
-    fetchRequest?.(pageSize, pageIndex);
-  }, [pageIndex, pageSize]);
+  const handleChangePage = (event: React.MouseEvent<HTMLButtonElement> | null, newPage: number) => {
+    setPage && setPage(newPage);
```
Comment (ZmicerBoksha) [inline src/components/common/table/tablePagination/tablePagination.tsx]:
```suggestion
    setPage?.(newPage);
```

File: src/components/common/table/tablePagination/tablePagination.tsx [hunk 3]
```diff
@@ -13,43 +12,43 @@ const useStyles = makeStyles(() => {
...
 
-  const handleChangeRowsPerPage = useCallback(
-    event => {
-      setPageSize(Number(event.target.value));
-    },
-    [setPageSize],
-  );
+  const handleChangeRowsPerPage = (event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
+    setItemsPerPage && setItemsPerPage(parseInt(event.target.value, 10));
+    setPage && setPage(0);
```
Comment (ZmicerBoksha) [inline src/components/common/table/tablePagination/tablePagination.tsx]:
```suggestion
    setPage?.(0);
```

File: src/components/common/table/tablePagination/tablePagination.tsx [hunk 4]
```diff
@@ -13,43 +12,43 @@ const useStyles = makeStyles(() => {
...
+  };
 
-  const handleChangeRowsPerPage = useCallback(
-    event => {
-      setPageSize(Number(event.target.value));
-    },
-    [setPageSize],
-  );
+  const handleChangeRowsPerPage = (event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
+    setItemsPerPage && setItemsPerPage(parseInt(event.target.value, 10));
```
Comment (ZmicerBoksha) [inline src/components/common/table/tablePagination/tablePagination.tsx]:
```suggestion
    setItemsPerPage?.(parseInt(event.target.value, 10));
```

File: src/components/events/events.tsx [hunk 5]
```diff
@@ -38,25 +39,44 @@ const Events: FunctionComponent = () => {
...
+      .getEvents(page, itemsPerPage, searchParam)
       .then(response => {
         setData(response.content);
+
+        return response;
+      })
+      .then(response => {
+        setCountRows(response.totalElements);
+        return response;
       })
```
Comment (ZmicerBoksha) [inline src/components/events/events.tsx]: can we combine these two then?
`.then(response => {
        setData(response.content);
        setCountRows(response.totalElements)
        return response;
      })`

Comment (ArteGon) [reply src/components/events/events.tsx]: Yes, I'll do it

**Review by DmitryPateew — APPROVED** [review]

**Review by ZmicerBoksha — APPROVED** [review]

**Review by vchepurnaya — APPROVED** [review]

**Review by DmitryPateew — APPROVED** [review]

**Pull request merged**